lib/lightstep/tracer/span.rb

```diff
@@ -0,0 +1,132 @@
+# TODO(ngauthier@gmail.com) Separate Span and SpanContext under a getter according
+# to the spec. Baggage moves to span context.
+module LightStep
+  # Span represents an OpenTracer Span
+  #
+  # See http://www.opentracing.io for more information.
+  class Span
+    # UnsupportedValueTypeError is raised when a tag value is not a supported type
+    class UnsupportedValueTypeError < LightStep::Error; end
+
+    # The guid of the span
+    attr_reader :guid
+    # Tags on the span
+    attr_reader :tags
+    # The baggage attached to this span
+    attr_reader :baggage
+    # The {Tracer} that created this span
+    attr_reader :tracer
+
+    # The guid of the current trace
+    attr_accessor :trace_guid
+    # The operation name
+    attr_accessor :operation_name
+    # Start time of the span in microseconds
+    attr_accessor :start_micros
+    # End time of the span in microseconds
+    attr_accessor :end_micros
```

Comment:
(same comment as above about keeping these accessors out of the public API we need to support going forward)

For this line in particular, `finish_micros` would be more consistent with OT naming.